### ISSUE TITLE
segfault on printing Integral of phi(t)

### ISSUE THREAD

**Issue by cbm755** [sympy/sympy#9057]
```
In [1]: from sympy import *
In [2]: from sympy.abc import phi, t
In [3]: e0 = Integral(phi(t), (t, 0, 1))

In [4]: pprint(e0)

1        
⌠        
⎮ φ(t) dt
⌡        
0        

In [5]: q = e0 + 2

In [6]: print(q)

Segmentation fault (core dumped)
```

This causes the test in #8416 / #6134 to fail (segfault)

**Comment by cbm755** [comment]
I've traced this failure to:
````
e0 = integrate(phi(t),(t,0,1))
e1 = integrate(alpha(t),(t,0,1))
e1.evalf()   # TypeError: can't convert expression to float
e0.evalf()   # segfault
````

Then further to:
````
phi = Symbol('phi')
phi(0.5)     # segfault
````

Evaluating Symbols at Floats seems to misbehave: `beta` and `phi` behave differently than `alpha`.  Of course `beta` *is* the beta function by default.  But it looks like it becomes such again even when overwritten with `beta = Symbol('beta')`.

This looks suspicious and is invoked by evalf:
````
    155     def __call__(self, *args):
    156         from .function import Function
--> 157         return Function(self.name)(*args)
````

**Comment by cbm755** [comment]
Ok, self contained example.  Consider:
````
beta = Symbol('beta')
beta(2)             # output is "beta(2)"
beta(2.5)
````
But the last one outputs:
````
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-45022b875f5a> in <module>()
----> 1 beta(2.5)

/usr/lib/python2.7/site-packages/sympy/core/symbol.pyc in __call__(self, *args)
    140     def __call__(self, *args):
    141         from .function import Function
--> 142         return Function(self.name)(*args)
    143 
    144     def as_real_imag(self, deep=True, **hints):

/usr/lib/python2.7/site-packages/sympy/core/function.pyc in __new__(cls, *args, **options)
    683     def __new__(cls, *args, **options):
    684         args = list(map(sympify, args))
--> 685         obj = super(AppliedUndef, cls).__new__(cls, *args, **options)
    686         return obj
    687 

/usr/lib/python2.7/site-packages/sympy/core/function.pyc in __new__(cls, *args, **options)
    380         pr2 = min(cls._should_evalf(a) for a in result.args)
    381         if pr2 > 0:
--> 382             return result.evalf(mlib.libmpf.prec_to_dps(pr))
    383         return result
    384 

/usr/lib/python2.7/site-packages/sympy/core/evalf.pyc in evalf(self, n, subs, maxn, chop, strict, quad, verbose)
   1315             options['quad'] = quad
   1316         try:
-> 1317             result = evalf(self, prec + 4, options)
   1318         except NotImplementedError:
   1319             # Fall back to the ordinary evalf

/usr/lib/python2.7/site-packages/sympy/core/evalf.pyc in evalf(x, prec, options)
   1215             if 'subs' in options:
   1216                 x = x.subs(evalf_subs(prec, options['subs']))
-> 1217             re, im = x._eval_evalf(prec).as_real_imag()
   1218             if re.has(re_) or im.has(im_):
   1219                 raise NotImplementedError

/usr/lib/python2.7/site-packages/sympy/core/function.pyc in _eval_evalf(self, prec)
    484 
    485         with mpmath.workprec(prec):
--> 486             v = func(*args)
    487 
    488         return Expr._from_mpmath(v, prec)

TypeError: beta() takes exactly 3 arguments (2 given)
````

**Comment by cbm755** [comment]
Maybe this is all just a dupe of #6134?  The test isn't valid because it doesn't pass, it just segfaults.  

Also, as far as I can tell, this test has been segfaulting for a long time but it does not seem to cause the tests to be marked failing (!)

**Comment by asmeurer** [comment]
Unless you are using compiled code or ctypes (SymPy does neither), a segfault is a bug in the Python interpreter. What is your exact `sys.version`?

**Comment by cbm755** [comment]
````
sys.version
Out[4]: '2.7.8 (default, Nov 10 2014, 08:19:18) \n[GCC 4.9.2 20141101 (Red Hat 4.9.2-1)]'
sympy.__version__
Out[5]: '0.7.6'
````
This is on Fedora 21.

**Comment by cbm755** [comment]
I cannot reproduce the segfault anymore so I will close this.  The underlying cause is fixed by #9064 (which has its own issue number, #3539).
